lib/service/task_service.dart

```diff
@@ -1,19 +1,26 @@
+import 'dart:convert';
 import 'package:task_list_app/client/backend_client.dart';
+import 'package:task_list_app/models/task.dart';
 
 class TaskService {
-  const TaskService();
+  TaskService();
   static const String taskApiBase = '/tasks';
+  BackendClient client = const BackendClient();
 
-  Future<void> addTask({required String taskName}) async {}
+  Future<void> addTask({required Task taskName}) async {}
 
-  Future<String> readTasks() async {
-    BackendClient client = const BackendClient();
-    return await client.getData(uri: taskApiBase);
+  Future<List<Task>> readTasks() async {
+    String apiTask = await client.getData(uri: taskApiBase);
+    dynamic data = jsonDecode(apiTask);
```

Comment:
I think I would recommend doing the `jsonDecode()` in the client. Decoding the JSON seems like a very REST API / communication with the back-end sort of thing.

I think that would leave this as:

```suggestion
    dynamic apiTask = await client.getData(uri: taskApiBase);
```

Reply:
makes sense, I was wondering about that too